### ISSUE TITLE
Not working on this website

### ISSUE THREAD

**Issue by Leioa2** [richardfrost/AdvancedProfanityFilter#330]
motherless.com loads forever while this extension is enabled, Tested on Chrome and Firefox both have the same issue.

**Comment by Leioa2** [comment]
Hello again! I'm sorry for being too late. My list contains 80+ words so it took me a lot of time to research, I noticed that the website loads faster the more words are deleted, So I guess the problem is related with the massive number of words, But the same list works fine on every other website and loads quickly, So I think the website itself is not optimized enough.

**Comment by richardfrost** [comment]
No worries, that is quite the list of words to go through. Are you using the normal mode (as opposed to "Advanced" or "Deep") on the site in question? The number of words in the list does impact performance, but usually its not quite that severe. Its possible that the site isn't very optimized like you say, or that they are using one of the words in your list for an internal part of their site that is triggering extra overhead.

So does the site eventually load with your full word list?

**Comment by Leioa2** [comment]
I'm using normal mode for sure, I've also checked every word, none of my words caused the issue, the site eventually load with full word list If I wait 2 minutes or more, I have to wait again if I refresh the page or transferred to another section, And that's really annoying.

I tried to restore the default settings and export it as a .json file to see if the default settings are different from mine, I compared the two json files through notepad, It appears to me that no difference between the two settings, they all match.

It can only be explained that the number of words was triggering the loading delay, I can see Firefox is stuck on "Transferring data..." during the loading process with the extension enabled, I found out that the website is the one to blame. Thank you so much for your response.

**Comment by richardfrost** [comment]
No worries! Thanks for doing such a deep dive to compare those config changes. It does indeed sound like the issue is just how many words are there. One option you have is creating a smaller [group of words (wordlist)](https://github.com/richardfrost/AdvancedProfanityFilter/wiki#wordlists) for slow sites like the one you mentioned. That way you can still get some filtering if you'd like while still keeping the website usable.

Basically you could create a minimal list that only has 15-20 words on it, and set that site to use that wordlist. Let me know if I can help any more, or feel free to close this if you feel like you are good with the way things are. Thanks again for reporting the issue and taking so many steps to help troubleshoot it.

**Comment by Leioa2** [comment]
The wordlist sounds a good idea! It will reduce the lag a lot, I appreciate your quick feedback. <3
